### DISCUSSION TITLE
Use Puppet without server?

### DISCUSSION THREAD

**Discussion by Sigurdur42** [puppetlabs/community#67] [General]
Hi folks,
I have had a brief look into puppet and really like the idea :)

I am thinking on managing my local Linux machine settings with it - as I am kind of a distro hopper, puppet should enable me to have a new machine up and running in no time. Only downside here is: I do not want to set up a local puppet server - I only want to use puppet using puppet apply to avoid the additional infrastructure.

Is there a good guide how this can be achieved? I tied with a local repo but it seems that hiera is not properly evaluated which makes me think, I am missing something fundamental...

Thx
Michael

P.S.: You can also talk me out of the idea if you think this is totally idiotic :)

**Comment by binford2k** [comment]
As long as you're using hiera properly and have configured any non-default datasources you want to use, it should just work. What distribution and what Puppet version?

```
[~]$ cat ~/.puppetlabs/etc/puppet/data/common.yaml
message: cat scrratch fever

[~]$ puppet apply -e 'notice(lookup("message"))'
Notice: Scope(Class[main]): cat scrratch fever
Notice: Compiled catalog for arachne.local in environment production in 0.03 seconds
Notice: Applied catalog in 0.01 seconds
```
